platform/mbed_wait_api_rtos.cpp

```diff
@@ -32,15 +33,19 @@ void wait_ms(int ms) {
 }
 
 void wait_us(int us) {
-    uint32_t start = us_ticker_read();
+    const ticker_data_t *const ticker = get_us_ticker_data();
+
+    uint32_t start = ticker_read(ticker);
     // Use the RTOS to wait for millisecond delays if possible
     int ms = us / 1000;
     if ((ms > 0) && core_util_are_interrupts_enabled()) {
+        sleep_manager_lock_deep_sleep();
```

Comment:
The current implementation of wait doesn't allow deep sleep, since the us ticker must be running. Two possible solutions to this are:
* Make distinction between busy wait and suspended wait (encourage use of `Thread::wait` again)
* Use `Thread::wait` for ms part and busy wait for remainder (start time would need to be read after `Thread::wait`)

Anyone have a preference or other ideas on how to handle this?

Reply:
> The current implementation of wait doesn't allow deep sleep, since the us ticker must be running.

tickers are always running (only initialize function present), are we getting API to stop/resume them?

> Make distinction between busy wait and suspended wait (encourage use of Thread::wait again)

From a user perspective, what API would be preferable ? 

If I consider the second one (ms part and busy), what about these:

`wait_us()` - Busy wait with higher accurancy (microseconds) always blocking, no Thread:wait involved. We are specifically requesting high precision and be blocking. Smaller periods (limits to 32bit microseconds).
`wait_ms()` + `wait()` - lower accuracy (task switch allowed), plus tickless possible. Bigger periods.

Reply:
Largely agree with this proposal - see #5216.

Only caveat is that it is something of an API change - wait_ms will wait up to that number of milliseconds, being potentially low by up to a millisecond, rather than waiting at least that number of milliseconds.

cf confusion when RTX changed a similar function: http://www.keil.com/support/docs/3766.htm and linked threads.

Although in this case the change would be towards conventional/expected behaviour, rather than away from.